.github/workflows/build_idf_examples.yml

```diff
@@ -11,18 +11,38 @@ jobs:
     strategy:
       matrix:
         idf_ver: ["release-v5.0", "release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "latest"]
+        target: ["esp32s2", "esp32s3", "esp32p4"]
+        example:
+          - { name: "USB Device", path: "examples/peripherals/usb/device", manifest_path: "examples/peripherals"}
+          - { name: "Network", path: "examples/network", manifest_path: "examples/network" }
```

Comment:
This creates another diminsion of the matrix, now we have 23 jobs for simple example builds, this should be optimized

number of jobs: idf_ver * target * example

![image](https://github.com/user-attachments/assets/930349e6-94bf-44f5-8481-038580cbe696)